### PR TITLE
nodeport: nat map size and collision retries

### PR DESCRIPTION
**Pull request by borkmann**
/cc @brb

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8850)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

**Comment by borkmann** [comment]
test-me-please

File: Documentation/cmdref/cilium-agent.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ cilium-agent [flags]
       --bpf-compile-debug                          Enable debugging of the BPF compilation process
       --bpf-ct-global-any-max int                  Maximum number of entries in non-TCP CT table (default 262144)
       --bpf-ct-global-tcp-max int                  Maximum number of entries in TCP CT table (default 1000000)
+      --bpf-nat-global-max int                     Maximum number of entries for the global BPF NAT table (default 841429)
```
Comment (aanm) [inline Documentation/cmdref/cilium-agent.md:28]: that's a really odd default where did that number came from?

File: pkg/maps/nat/nat.go [hunk 2]
```diff
@@ -103,7 +101,7 @@ func NewMap(name string, v4 bool) *Map {
 			sizeKey,
 			mapValue,
 			sizeVal,
-			MaxEntries,
+			option.Config.NATMapEntriesGlobal,
```
Comment (aanm) [inline pkg/maps/nat/nat.go]: make this as an argument of `NewMap`

File: pkg/option/config.go [hunk 3]
```diff
@@ -384,6 +384,12 @@ const (
 	CTMapEntriesGlobalTCPName    = "bpf-ct-global-tcp-max"
 	CTMapEntriesGlobalAnyName    = "bpf-ct-global-any-max"
 
+	// NATMapEntriesGlobalDefault holds the default size of the NAT map
+	NATMapEntriesGlobalDefault = int((CTMapEntriesGlobalTCPDefault + CTMapEntriesGlobalAnyDefault) * 2 / 3)
```
Comment (aanm) [inline pkg/option/config.go:389]: Why the `*2/3`?

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by brb — APPROVED** [review]
LGTM!

File: Documentation/cmdref/cilium-agent.md (hunk 1 above)
Comment (borkmann) [reply Documentation/cmdref/cilium-agent.md:28]: We've discussed to use a heuristic to be 2/3 of the total CT map size by default, see also commit description.

File: pkg/maps/nat/nat.go (hunk 2 above)
Comment (borkmann) [reply pkg/maps/nat/nat.go]: Makes sense, will do

**Comment by borkmann** [comment]
test-me-please

File: pkg/option/config.go (hunk 3 above)
Comment (aanm) [reply pkg/option/config.go:389]: @borkmann but it should be as a comment here not only in the commit

Comment (borkmann) [reply pkg/option/config.go:389]: Ok, done

**Comment by borkmann** [comment]
test-me-please

**Pull request merged**